### PR TITLE
Add an option to configure `WDT` action

### LINKED ISSUE #1831: There is currently no way to configure WDT to do anything but a system reset
While we can set an interrupt handler, we currently have no way for the user to configure the WDT to cause anything but a system reset.

### PR DESCRIPTION
**Pull request by playfulFence**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [x] I have added necessary changes to user code to the [Migration Guide](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/MIGRATING-0.21.md).
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/documentation/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/documentation/CONTRIBUTING.md) and followed its instructions.

#### Description
close #1831

Also made some small improvements in code quality, aligned mode names with the official Espressif documentation

Currently, setting up MWDT stages is only possible with custom bootloader. Also, I think we should document the whole process of creating a custom bootloader and using it with our driver (I've opened an issue for this: https://github.com/esp-rs/esp-hal/issues/2424)

#### Testing
Used implemented function in example.

### CODE REVIEW

File: esp-hal/src/rtc_cntl/mod.rs [hunk 1]
```diff
@@ -778,36 +778,46 @@ impl RtcClock {
...
 
 /// RTC Watchdog Timer driver.
 impl Rwdt {
+    /// Create a new RTC watchdog timer instance
+    pub fn new() -> Self {
+        Self
+    }
+
     /// Enable the watchdog timer instance.
+    /// Watchdog starts with default settings (`stage 0`` resets the system, the
```
[the remark this answers is not in the document]
Comment (playfulFence) [reply esp-hal/src/rtc_cntl/mod.rs:819]: Oops, thanks, will be fixed 😄

**Comment by playfulFence** [comment]
Update: I've managed to resolve problems with `MWDT`, it was about bootloader settings. 
It turns out we need to disable `TASK_WDT` and `INT_WDT` to make `MWDT` work "predictable". 
I'm working on a proper solution of how to implement it in our driver in a less painful way...

**Comment by jessebraham** [comment]
We discussed this in our triage meeting today. I think for the time being, we are okay with simply documenting the requirements for the bootloader; we'd like to continue investigating if we can bypass this step, however there's no reason not to move forward with this PR IMO.

@playfulFence when you're able, could you please make sure the requirements for MWDT are clearly documented, and then get this PR ready for review/merge?

File: esp-hal/CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 
 ### Added
+Add an option to configure `WDT` action (#2330)
```
Comment (bjoernQ) [inline esp-hal/CHANGELOG.md:11]: this line looks a bit out of place and is also missing the leading dash

File: esp-hal/src/rtc_cntl/mod.rs [hunk 3]
```diff
@@ -914,57 +920,160 @@ impl Rwdt {
...
 
         self.set_write_protection(true);
     }
 
-    /// Configure timeout value in ms.
-    pub fn set_timeout(&mut self, timeout: MicrosDurationU64) {
+    /// Configure timeout value in ms for the selected stage.
+    pub fn set_timeout(
+        &mut self,
+        stage: usize,
```
Comment (bjoernQ) [inline esp-hal/src/rtc_cntl/mod.rs:961]: Having an enum for the stage would be better - we could also remove that one error then

**Review by bjoernQ — APPROVED** [review]
LGTM

**Comment by bjoernQ** [comment]
While we can't test the reset functionality we could probably have HIL tests to test the interrupt - but maybe in a follow-up PR

**Review by jessebraham — APPROVED** [review]
Thanks, LGTM! (Other than the broken imports 😁 )

**Pull request merged**